daft/dataframe/dataframe.py

```diff
@@ -3622,7 +3622,7 @@ def map_groups(self, udf: Expression) -> "DataFrame":
             >>>
             >>> @daft.udf(return_dtype=daft.DataType.float64())
             ... def std_dev(data):
-            ...     return [statistics.stdev(data.to_pylist())]
+            ...     return [statistics.stdev(data)]
```

Comment:
We get a marginal improvement in memory-efficiency here. Originally:
![7A3A48EC-1440-47CF-BD73-186F4DEC1970](https://github.com/user-attachments/assets/0a4ac43f-d505-4249-9e92-5c0a189caee0)

After using the iterator:
![90409482-39C5-44A6-9C9E-00167D0765E9](https://github.com/user-attachments/assets/ccf2bfec-d050-44cf-b26e-ad728161ecbe)

Reply:
Heap size spike only happens on macOS due to the OS not recovering freed heap allocations. On linux the heap size follows RSS quite closely for both cases.